pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -183,6 +183,11 @@ async def _prepare_messages(
 
         if user_prompt is not None:
             parts.append(_messages.UserPromptPart(user_prompt))
+        elif len(parts) == 0 and message_history and isinstance(message_history[-1], _messages.ModelRequest):
```

Comment:
Type wise, if you check `isinstance(messages[-1], _messages.ModelRequest)`, then you don't need to cast below.